### PR TITLE
Fixing labels on GKE clusters

### PR DESCRIPTION
**Pull request by vrutz**
Inheritance is not added yet. We are populating Kubernetes labels and not GCP labels. To be done in another PR (https://app.shortcut.com/dataiku/story/173304/gke-ability-to-add-label-and-taint-on-node)

### CODE REVIEW

**Comment by amandineslx** [comment]
[sc-194897]

File: python-clusters/create-gke-cluster/cluster.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def start(self):
                 node_pool_builder.with_auto_scaling(node_pool.get('numNodesAutoscaling', False), node_pool.get('minNumNodes', 2), node_pool.get('maxNumNodes', 5))
                 node_pool_builder.with_gpu(node_pool.get('withGpu', False), node_pool.get('gpuType', None), node_pool.get('gpuCount', 1))
                 node_pool_builder.with_spot_vms(node_pool.get('useSpotVms', False))
-                node_pool_builder.with_nodepool_labels(node_pool.get('nodepoolLabels', {}))
+                node_pool_builder.with_nodepool_labels(node_pool.get("nodepoolLabels", {}))
```
Comment (amandineslx) [inline python-clusters/create-gke-cluster/cluster.py:64]: should we wait for a Python style PR to change them all?

File: python-lib/dku_google/clusters.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 from dku_google.gcloud import get_instance_network, get_instance_service_account
 from dku_utils.access import _has_not_blank_property, _is_none_or_blank, _default_if_blank, _merge_objects
 
-import os, sys, json, re
+import os, sys, json, re, random
```
Comment (amandineslx) [inline python-lib/dku_google/clusters.py:8]: `random` import does not seem to be used

File: python-lib/dku_google/clusters.py [hunk 3]
```diff
@@ -232,6 +232,7 @@ def with_initial_node_count(self, node_count):
         return self
     
     def with_labels(self, labels={}):
+        labels = {label["from"]: label.get("to", "") for label in labels}
```
Comment (amandineslx) [inline python-lib/dku_google/clusters.py:235]: do we want to raise an error explicitly if the label key (from) is empty?

File: python-lib/dku_google/clusters.py [hunk 4]
```diff
@@ -110,7 +110,7 @@ def with_spot_vms(self, use_spot_vms):
 
     def with_nodepool_labels(self, nodepool_labels=[]):
         if nodepool_labels:
-            nodepool_labels_dict = {l["from"]: l["to"] for l in nodepool_labels}
+            nodepool_labels_dict = {label["from"]: label.get("to", "") for label in nodepool_labels}
```
Comment (amandineslx) [inline python-lib/dku_google/clusters.py:113]: do we want to raise an error explicitly if the label key (from) is empty?

**Review by amandineslx — CHANGES_REQUESTED** [review]
See comments
Functionally speaking, it works

### Tested

* deploy DSS in GCP with FM through an elastic fleet
* update plugin with the one built from the PR (commit `ac9a36126e5ac8003d73249963817c238af96242`)
* ❓ create a cluster with labels with empty key &rarr; error raised
* ❓ create a cluster with a node group label with empty key &rarr; error raised
* ✅ create a cluster with labels with empty value and both key and value &rarr; checked in Google cloud console
* ✅ create a cluster with a node group with label with empty value and both key and value &rarr; checked in Google cloud console

File: python-lib/dku_google/clusters.py (hunk 2 above)
Comment (vrutz) [reply python-lib/dku_google/clusters.py:8]: It is used in a method that's not part of this PR. Not sure it's ever really called, but I thought it'd be harmless to add that import.
The link to the line: https://github.com/dataiku/dss-plugin-gke-clusters/blob/ac9a36126e5ac8003d73249963817c238af96242/python-lib/dku_google/clusters.py#L259

File: python-clusters/create-gke-cluster/cluster.py (hunk 1 above)
Comment (vrutz) [reply python-clusters/create-gke-cluster/cluster.py:64]: I think that makes sense. Reverting that now.

File: python-lib/dku_google/clusters.py (hunk 3 above)
Comment (vrutz) [reply python-lib/dku_google/clusters.py:235]: So apparently, the form will not let you save an entry with no value. If you try to cheat and put multiple entries, one without key and one with, the entry without value will disappear and not be saved.
So I don't think we can really end up in that situation, but some defensive coding can't hurt

File: python-lib/dku_google/clusters.py [hunk 5]
```diff
@@ -232,6 +235,10 @@ def with_initial_node_count(self, node_count):
         return self
     
     def with_labels(self, labels={}):
+        if any(labels, lambda label: not label.get("from", "")):
+            raise ValueError("Some of the cluster key-value label pairs have no key and thus are invalid: %s" % labels)
```
Comment (amandineslx) [inline python-lib/dku_google/clusters.py:239]:
```suggestion
            raise ValueError("Some of the cluster key-value label pairs have no key and thus are invalid")
```

**Review by amandineslx — CHANGES_REQUESTED** [review]
See comments

### Tested
* deploy DSS in GCP with FM through an elastic fleet
* update plugin with the one built from the PR (commit `138ece5fa2814d67b381b88c1cbd93f3f0be825d`)
* ✅ create a cluster with labels with empty key &rarr; checked in Google cloud console
* ✅ create a cluster with a node group label with empty key → checked in Google cloud console
* ✅ create a cluster with labels with empty value and both key and value → checked in Google cloud console
* ✅ create a cluster with a node group with label with empty value and both key and value → checked in Google cloud console
* ✅ add a node group through the macro with label with empty value and both key and value → checked in Google cloud console

File: python-lib/dku_google/clusters.py [hunk 6]
```diff
@@ -232,6 +235,10 @@ def with_initial_node_count(self, node_count):
         return self
     
     def with_labels(self, labels={}):
+        if any(labels, lambda label: not label.get("from", "")):
```
Comment (vrutz) [inline python-lib/dku_google/clusters.py:238]:
```suggestion
        if any(not label.get("from", "") for label in labels):
```

**Review by amandineslx — APPROVED** [review]
OK for me with the latest changes

**Pull request merged**